packages/services/src/kernel/comm.ts

```diff
@@ -212,6 +214,31 @@ export class CommHandler extends DisposableDelegate implements Kernel.IComm {
     return future;
   }
 
+  dispose(): void {
+    super.dispose();
+
+    this._maybeCloseSubshell();
+  }
+
+  private async _maybeStartSubshell() {
+    if (this._kernel.supportsSubshells) {
```

Comment:
There should be an `await this._kernel.info` before this, as we can only tell if a kernel supports subshells after we've received the kernel info which contains the appropriate information.